### DISCUSSION TITLE
Expose ChainModal in API

### DISCUSSION THREAD

**Discussion by 0xcadams** [rainbow-me/rainbowkit#381] [Ideas]
Hey all, thanks for creating this repo and excited to be an early adopter w/ @livepeer!

We're integrating rainbowkit and looking to have our users forced to switch to the correct network before using the app, to prevent issues with interacting with the wrong network. It would be great to either have a `forceNetwork` prop for the ConnectButton or a way to customize the ChainModal text to be able to add more detail as to why the modal is open. Either way, I'd love to contribute to this repo so I can open a PR depending on input.

**Comment by hzhu** [comment]
+1, I think having an option to expose the chain dropdown + modal somehow would be nice. 

A dApp may allow a user to interact with the app that involves a `chainId` before wallet connection. My API operates on a `chainId` and does not require the user to have a connected wallet, _initially_.  For example, here's Uniswap's connect button in a unconnected state:

<img width="364" alt="Screen Shot 2022-06-19 at 11 32 15 PM" src="https://user-images.githubusercontent.com/1811365/174539324-e554529e-c1e4-4591-b5c5-8367b0705605.png">

**Comment by markdalgleish** [comment]
[RainbowKit v0.4.3](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%400.4.3) just shipped with a couple of features that are relevant to this:
- We now provide Hooks for programmatically opening the modals: https://www.rainbowkit.com/docs/modal-hooks, e.g.
  ```tsx
  import { useChainModal } from "@rainbow-me/rainbowkit";
  
  export default () => {
    const { openChainModal } = useChainModal();
  
    return (
      <button disabled={!openChainModal} onClick={openChainModal} type="button">
        Open chain modal
      </button>
    );
  };
  ```
- We allow you to specify the initial chain to connect to via the `initialChain` prop on `RainbowKitProvider`, e.g.
   ```tsx
   <RainbowKitProvider chains={chains} initialChain={chain.mainnet}>
   ```
   
   or
   
   ```tsx
   <RainbowKitProvider chains={chains} initialChain={1}>
   ```

**Reply by 0xcadams** [reply]
Oh wow this is perfect! Thanks!

**Comment by oxSaturn** [comment]
Seems wagmi v2 will allow switching network without connected first, might be good timing to reconsider this feature request.
